### ISSUE TITLE
Can't disable tray icon on Windows

### ISSUE THREAD

**Issue by upthekhyber** [flameshot-org/flameshot#2774]
### Flameshot Version

12.1.0

### Installation Type

Using the ready-made package from Github Releases

### Operating System type and version

Windows 10 Enterprise 21H2 (64-bit)

### Description

I am unable to disable Flameshot's tray icon by setting `disabledTrayIcon=true` on Windows.
Has this not been implemented in the Windows version by any chance? Any workarounds?

### Steps to reproduce

1. Exit flameshot.exe
2. Open `%AppData%\flameshot\flameshot.ini`
3. Set `disabledTrayIcon=true` under `[General]`
4. Start flameshot.exe
5. Tray icon is still showing up

### System Information

OS: Windows 10 Enterprise 21H2 (64-bit)
Monitor configuration: tested w/ various setups (single 1920x1080, dual 2560x1440, etc.)

**Comment by manolol1** [comment]
I've also tried that, and it starts completely normal (like if it's just ignoring that flag).

OS: Windows 11 Pro 21H2 (64-bit)

**Comment by fernandofig** [comment]
Looking here: https://github.com/flameshot-org/flameshot/blob/f7e41f4d708e50eeaec892408069da25a28e04a2/src/core/flameshotdaemon.cpp#L332

It seems the option is completely ignored on Windows - it'll always create a tray icon. Glancing on some other parts of that source, this option also seems tied to a GUI setting that's hidden / not applicable on Windows builds.

It would seem easy enough to shuffle around those ifdefs and code blocks so the Windows build would respect that setting, but I have no idea if there's any behavior downstream that's specific to the the Windows build that depends on the tray icon, that would consequently break if the FlameshotDaemon::initTrayIcon method would be changed that way.

The compile instructions seems lacking on the Windows front - I'm not sure if that's because it's straightforward to the point of not needing anything but the dependencies, or it's really overlooked. On top of all that, I'm not really a C++ guy, so it would be better if more seasoned developers would have a look at this, at least to assess the difficulty in fixing this problem. @mmahmoudian ?

**Comment by mmahmoudian** [comment]
> it would be better if more seasoned developers would have a look at this, at least to assess the difficulty in fixing this problem. @mmahmoudian ?

I agree, I'm not that seasoned either, so here are people who know better: @veracioux @ZetaoYang @hosiet
